### ISSUE TITLE
Misleading return type definition for table.get_item()

### ISSUE THREAD

**Issue by pavlobielousov** [youtype/mypy_boto3_builder#241]
**Describe the bug**
The current TypedDict definition for the DynamoDB table `get_item()` method `GetItemOutputTableTypeDef` contains "Item" as a required key, however, this is not true. According to the [AWS documentation](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_GetItem.html):

> If there is no matching item, `GetItem` does not return any data and there will be no `Item` element in the response.

Currently, `mypy` reports any branch of code that questions the existence of the "Item" key as an unreachable code.

**To Reproduce**

1. Install `boto3-stubs[dynamodb]`
2. Run `mypy` on the following code sample

```python
import boto3

db = boto3.resource("dynamodb")
table = db.Table("table-name")
response = table.get_item(Key={"foo": "bar"})

if "Item" in response:
    ...
else:
    # unreachable code according to mypy
    ...
...
```

**Actual output**

```
.../main.py:10: error: Statement is unreachable  [unreachable]
```

**Expected output**

No `mypy` reported errors.

**Additional context**
Windows 11, `boto3-stubs` installatalled via pipenv, `boto3` version 1.34.19

**Comment by vemel** [comment]
Hello! Thank you for the report. I am currently working on optional keys in response StructureShapes. This looks related to #238

**Comment by vemel** [comment]
The issue has finally been fixed in `mypy_boto3_builder 7.24.0` release. I have also released `mypy-boto3-dynamodb 1.34.91` with the fix included. Please update and let me know if it works as expected.
